crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -208,10 +286,18 @@ impl ServerBuilder {
         Ok(cors)
     }
 
+    /// Consumes the `ServerBuilder` to create a `Server` that can be run.
     pub fn build(self) -> Result<Server, Error> {
-        let (address, schema, router) = self.build_components();
+        let state = self.state.clone();
+        let (address, schema, db_reader, router) = self.build_components();
+
+        // Start the background task to update a shared `CheckpointViewedAt`
+        let watermark = Arc::new(AtomicU64::new(0));
```

Comment:
This type (`Arc<AtomicU64>`) is too generic to be made into an extension all on its own -- consider wrapping it in some more meaningful type that makes it clear what this number represents.

Reply:
would something like
```
/// The high checkpoint watermark stamped on each GraphQL request. This is used to ensure
/// cross-query consistency.
pub(crate) type CheckpointWatermark = Arc<AtomicU64>;
```
suffice?

Reply:
Well it needs to be a `struct` rather than a type alias, but `struct CheckpointWatermark(Arc<AtomicU64>)` would suffice. Type aliases are not generative, meaning if you used `CheckpointWatermark` as you defined it, Rust would not view it as a different type to `Arc<AtomicU64>`, which is what we want here.